### PR TITLE
Pass DrawPixelInfo by reference in most places

### PR DESCRIPTION
**Pull request by Gymnasiast**
This refactors most of the places where DrawPixelInfo was passed as a pointer. The only places left out are the drawing engines, as I couldn’t get those working (they kept crashing on software rendering mode). Some else might perhaps want to take a look at those.

Blocked by https://github.com/OpenRCT2/OpenRCT2/pull/19693

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
https://github.com/OpenRCT2/OpenRCT2/pull/19693 looks to be almost done so I suggest we merge/rebase after that PR is merged.

**Review by tupaschoal — COMMENTED** [review]
Can you revert the ones on windows that have not been migrated yet, but are in progress?

I'd rather open an issue to track those to be done later than to have either you or them solving the merge conflicts, should be `Ride` and `GameBottomToolbar`

**Comment by duncanspumpkin** [comment]
> Can you revert the ones on windows that have not been migrated yet, but are in progress?
> 
> I'd rather open an issue to track those to be done later than to have either you or them solving the merge conflicts, should be `Ride` and `GameBottomToolbar`

I don't think that is possible as it modifies the consumers and producers. I suggest we just get it ready for merge right now and then get the other two PRs rebased as we don't know how long they will take to be ready.

**Comment by tupaschoal** [comment]
> I don't think that is possible as it modifies the consumers and producers. I suggest we just get it ready for merge right now and then get the other two PRs rebased as we don't know how long they will take to be ready.

Ah, ok

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**
